docs/dataset/object-detection.md

```diff
@@ -0,0 +1,366 @@
+    script in the code example for details on how a multiclass object detection dataset was generated and
+    an example for preparing a dataset file.
+
+=== "Web App"
+    To upload a dataset, having a properly formatted dataset file is a prerequisite.
+
+    To get started, navigate to [kolena.com](https://app.kolena.com/redirect/datasets) and
+    click `Import Dataset` then `Select From Cloud Storage`.
+    Using the explorer, navigate to `s3://kolena-public-examples/coco-2014-val/`
+    and select `coco-2014-val_person-detection.csv`. This is a dataset file for person detection using the COCO
+    dataset.
+
+    You will now see a preview of how the information is going to be consumed by Kolena.
+
+    Give your dataset a name and select `locator` as the ID field. The ID field uniquely identifies a datapoint
+    and is used when uploading model results to associate results with datapoints.
+
+    Click `Import` to create the dataset. Once the import has completed,
+    you can add descriptions and tags to organize your datasets.
+
+    <figure markdown>
+    ![COCO Dataset Upload](../assets/images/task-od-make-dataset-light.gif#only-light)
+    ![COCO Dataset Upload](../assets/images/task-od-make-dataset-dark.gif#only-dark)
+    <figcaption markdown>COCO Dataset Upload</figcaption>
+    </figure>
+
+=== "SDK"
+
+    The example code contains a script [`object_detection_2d/upload_dataset.py`](https://github.com/kolenaIO/kolena/blob/trunk/examples/dataset/object_detection_2d/object_detection_2d/upload_dataset.py)
+    which will process the CSV file `s3://kolena-public-examples/coco-2014-val/transportation/raw/coco-2014-val.csv`
+    and register a small transportation-based dataset in Kolena using the `register_dataset` function.
+
+    First, let's first configure our environment by populating the `KOLENA_TOKEN`
+    environment variable. Visit the [:kolena-developer-16: Developer](https://app.kolena.com/redirect/developer) page to
+    generate an API token and copy and paste the code snippet into your environment:
+
+    ```shell
+    export KOLENA_TOKEN="********"
+    ```
+
+    We can now register a new dataset using the provided script:
+
+    ```shell
+    poetry run python3 object_detection_2d/upload_dataset.py
+    ```
+
+    After this script has completed, a new dataset named `coco-2014-val` will be created,
+    and see in [:kolena-dataset-20: Datasets](https://app.kolena.com/redirect/datasets).
+
+### Step 2: Upload Object Detection Model Results
+
+Model results are supplied as tables containing the ID field chosen when uploading the dataset. For
+this example, we will upload object detection model results
+of [YOLO X](https://github.com/Megvii-BaseDetection/YOLOX) for the single class object detection task in the web app,
+and upload the results of [Faster R-CNN](https://github.com/facebookresearch/Detectron) for the
+multiclass object detection task through the SDK.
+
+??? info "Generating Model Results"
```

Comment:
Will [this](https://docs.kolena.com/dataset/advanced-usage/formatting-your-datasets/#formatting-results-for-object-detection) be an useful doc to link here?

Reply:
Also, what do you think of adding this as a separate step (a step before `Upload OD Model Results`)?